### PR TITLE
GameofLife homework

### PR DESCRIPTION
**Pull request by Lauren34**

### CODE REVIEW

File: src/main/java/lesser/gameoflife/GameofLife.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package lesser.gameoflife;
+
+public class GameofLife {
```
Comment (andrewoid) [inline src/main/java/lesser/gameoflife/GameofLife.java]: `GameOfLife`

File: src/test/java/lesser/gameoflife/GameofLifeTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package lesser.gameoflife;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class GameofLifeTest {
+
+    @Test
+    public void testInitialize() {
```
Comment (andrewoid) [inline src/test/java/lesser/gameoflife/GameofLifeTest.java]: `initialize()`

Tests methods should be named the name of the method that we are testing.

It's a very old convention to put the word "test" in the name of the test.

File: src/test/java/lesser/gameoflife/GameofLifeTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                }
+            }
+        }
+
+        assertTrue(hasAliveCells, "Grid should have some alive cells after initialization");
+        assertTrue(hasDeadCells, "Grid should have some dead cells after initialization");
+    }
+
+    @Test
+    public void testNextGen() {
```
Comment (andrewoid) [inline src/test/java/lesser/gameoflife/GameofLifeTest.java]: same as above

File: src/test/java/lesser/gameoflife/GameofLifeTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+        assertEquals(1, grid[0][1], "Cell (0,1) should be alive");
+        assertEquals(1, grid[1][1], "Cell (1,1) should be alive");
+        assertEquals(1, grid[2][1], "Cell (2,1) should be alive");
+        assertEquals(0, grid[1][0], "Cell (1,0) should be dead");
+        assertEquals(0, grid[1][2], "Cell (1,2) should be dead");
+    }
+
+    @Test
+    public void testSetCell() {
```
Comment (andrewoid) [inline src/test/java/lesser/gameoflife/GameofLifeTest.java]: same as above

File: src/main/java/lesser/gameoflife/GameofLife.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    public GameofLife(int width, int height) {
+        this.width = width;
+        this.height = height;
+        this.grid = new int[height][width];
+    }
+
+    public void initialize() {
+        for (int i = 0; i < height; i++) {
+            for (int j = 0; j < width; j++) {
+                grid[i][j] = (int) (Math.random() * 2);
```
Comment (andrewoid) [inline src/main/java/lesser/gameoflife/GameofLife.java]: `Math.randomBoolean() ? 1 : 0`

Comment (Lauren34) [reply src/main/java/lesser/gameoflife/GameofLife.java]: That line is showing up red if I try to replace it

Comment (andrewoid) [reply src/main/java/lesser/gameoflife/GameofLife.java]: What is the error of you try to replace it?

**Comment by Lauren34** [comment]
Would you be able to explain your comment?

**Comment by Lauren34** [comment]
[image: Screenshot 2024-09-11 at 10.45.26 AM.png]
This is the error and if I try to rename it it's still an error

**Comment by andrewoid** [comment]
Don't reply to the email, do your replies on the github website.

File: src/main/java/lesser/gameoflife/GameofLife.java (hunk 5 above)
Comment (Lauren34) [reply src/main/java/lesser/gameoflife/GameofLife.java]: Screenshot 2024-09-11 at 11 01 20 AM
this is the error

**Comment by Lauren34** [comment]
Sorry, just added the screenshot to github website

**Review by andrewoid — APPROVED** [review]

**Pull request merged**